ksiren/src/main/java/com/brightspace/ksiren/Action.kt

```diff
@@ -85,15 +106,24 @@ data class Action(
 			return obj
 		}
 	}
+}
 
-	fun hasField(name: String): Boolean {
-		this.fields.forEach {
-			if (it.name == name) {
-				return true
-			}
-		}
-		return false
-	}
+internal data class MutableAction(
+	override val name: String,
+	override val classes: List<String> = listOf(),
+	override val fields: List<MutableField> = listOf(),
+	override val href: String,
+	override val method: String = "GET",
+	override val title: String?,
+	override val type: ContentType = ContentType.FORM) : ActionBase {
```

Comment:
This feels excessive for only a single property being different, but I can't think of a better way to do it off the top of my head

Reply:
Yeah, especially given that this duplicates some default initializers. Ideally, the secondary constructor would be the primary constructor (that is, this can only accept an already-constructed `Action`/`ActionBase`), so I'll see I can make this more concise.

Reply:
Knee-jerk reaction suggestion:

```
internal class MutableAction(private val action: ActionBase) : ActionBase by action {
	val fieldValues: List<MutableField> = listOf()
}
```

Surprisingly, the compiler let me keep `data`, but I don't think you're making use of what that gives you, and I don't think it would work properly.

Reply:
So, this is exactly what I was looking for, but didn't run across/search for the correct things to see that delegation was the term used. Thanks!